# HG changeset patch
# User Rik <rik@octave.org>
# Date 1460069710 25200
#      Thu Apr 07 15:55:10 2016 -0700
# Node ID e3888aa9a4d178db9b65163fc982d532edc5ae5a
# Parent  fe5ed6d7fb10487b9c0f3bc36162f280fbe7561a
Restore tilde expansion for dlmread lost in fe5ed6d7fb10.

* dlmread.cc (Fdlmread): Use 'tname', the tilde expanded version of fname,
in the call to find_data_file_in_load_path.

diff --git a/libinterp/corefcn/dlmread.cc b/libinterp/corefcn/dlmread.cc
--- a/libinterp/corefcn/dlmread.cc
+++ b/libinterp/corefcn/dlmread.cc
@@ -210,17 +210,17 @@ such as text, are also replaced by the @
 
   if (args(0).is_string ())
     {
       // Filename.
       std::string fname (args(0).string_value ());
 
       std::string tname = file_ops::tilde_expand (fname);
 
-      tname = find_data_file_in_load_path ("dlmread", fname);
+      tname = find_data_file_in_load_path ("dlmread", tname);
 
       input_file.open (tname.c_str (), std::ios::in);
 
       if (! input_file)
         error ("dlmread: unable to open file '%s'", fname.c_str ());
 
       input = &input_file;
     }
